src/main/java/com/bootcamp/demo/Controller/CardController.java

```diff
@@ -0,0 +1,21 @@
+package com.bootcamp.demo.Controller;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.Firestore;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firebase.cloud.FirestoreClient;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/card")
+public class CardController {
+    private Firestore firestoreDB;
+
+    @DeleteMapping(value = "/{cardNumber}")
+    public String removeCard(@PathVariable String cardNumber){
+        this.firestoreDB = FirestoreClient.getFirestore();
+        ApiFuture<WriteResult> writeResult = firestoreDB.collection("cards").document(cardNumber).delete();
+        return "Successfully deleted " + cardNumber;
```

Comment:
If you want to take this to the next step you could make this method return a `ResponseEntity<String>` which will allow you to control the response status as well. The response status (and headers) can be used to handle exceptions (e.g. 404 in case the specified cardNumber is not found).
Note: This enhancement is optional for this exercise.

Reply:
I've changed it to return a ResponseEntity with the same message and HttpStatus.OK

Reply:
Sounds good! Just as a FYI `ResponseEntity` has a few builder methods that can help with building the response more easily.
`ResponseEntity.ok().body("body")` or `ResponseEntity.ok("body")`